### DISCUSSION TITLE
Ender 5+ and Silent Board V2.2 Wont Flash

### DISCUSSION THREAD

**Discussion by NextlvlEngineering** [Desuuuu/klipper#142] [Q&A] [answered]
I am not entirely new to the world of RPi, Arduino, and 3d printing in general, but I am struggling with getting my SB V2.2 (Silent Board V2.2) to flash klipper to it. I have configured the file in the Octopi  to the AVR/ATmega2650 and I am getting a ".elf.hex" as an output. Where I am struggling is that when I put the file on an SD and power up the printer nothing seems to be happening. Connecting the printer via USB I can still see that the board is still running the Creality configured Marlin V1.xx. I have attached the files output by the configuration tool. 

Unfortunately I also already flash my t5uid1 to the DGUS-reloaded firmware and cannot see what is happening and that loads to a blank screen after the Klipper logo goes away. The rest of the system still works and I can connect via USB and run the printer as usual, but the screen is still blank. I am assuming/hoping that once I get the mainboard flashed to Klipper that this screen will load as it should. 

Any help with this would be greatly appreciated.

**Comment by NextlvlEngineering** [comment, marked as answer]
Just tried to attach the output ".elf.hex" file but there does not seem to be a way to do so. I will throw it in a repo and post that link here so everyone can see it. 

Here is exactly what the document name is: 

"klipper.elf.hex"

I mostly have experience with STM32 based boards and those are typlically a ".bin" file I am assuming that this ".hex" is an ATMega equivalent of this bin file.

**Comment by UnderXirox** [comment]
Bonjour

Connecté au port USB
Du raspberry a l'USB de la carte mère de la machine

Taper ça dans le terminal du 
raspberry ou via le ssh
 

Type this in the terminal
 raspberry or via ssh 

sorry for my english

```
Path :

cd ~/
cd klipper/out/

sudo avrdude -cwiring  -patmega2560 -P/dev/ttyUSB0  -b115200 
-D -Uflash:w: klipper.elf.hex:i

```

`sudo avrdude -cwiring -patmega2560 -P/dev/ttyUSB0 -b115200 -D -Uflash:w:out/klipper.elf.hex:i`

**Reply by NextlvlEngineering** [reply]
Thank you for the help but I ended up reflashing my Pi with Mainsail instead of Octoprint and I was able to compile a hex file that worked. Not sure why it didnt work with octoprint, but I am good now thank you.
